web/html/src/components/FormulaForm.js

```diff
@@ -73,6 +73,11 @@ class FormulaForm extends React.Component {
                     metadata: {}
                 });
             else {
+                if (data.formula_list.filter(formula => formula == data.formula_name).length > 1) {
+                    this.state.errors.push(
```

Comment:
In current situation I consider it as an error. Merging data from multiple groups is silently disabled at the moment (see [uyuni/suma_minion.py#L155](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/modules/pillar/suma_minion.py#L155) ), and furthermore there is not a stable way to tell what group formula will "win" and be loaded.

However there is an exception for cluster-formula, @mbologna ?

Reply:
Right, there is an exception for that. We need to change the condition to allow multiple cluster formulas.

Reply:
Great, thanks for the clarification. It's clearer to me. :+1: 

(Based on the fact I knowing pretty much nothing about the issue itself in detail, I will leave here just a suggestion but feel free to ignore if not relevant/needed). As it is an error that needs to be fixed in order to continue, I'd enhance a little bit more the message text with an additional sentence about what should be done by the user explicitly. I mean something like `Please make sure to remove all but one "Bind" formula assigned to this system in order to...............`

Reply:
@ncounter the user does not need to disable any formula, we just ignore all but the first when we build the pillar data. The context of this change is that we have a user (SUMA customer) that expects that all group formulas are taken into account. Since I couldn't easily add those capabilities (prioritization when keys conflict and an unified UI that shows the merged data), we instead decided to make the current situation visible to the user. 

If this should be a warning or error is something I am not 100% sure though. For me either would be fine, but I want to be in line with the rest of Uyuni.

Reply:
I see. So generally speaking it should be:

a) `warning` = `continue` = `working, but keep in mind` = `hey Mr.Customer, be aware of this behavior you may do not expect it. The product will continue to work, but will do something like this instead of that`
b) `error` = `stop` = `not working` = `hey Mr.Customer, there is an error. You cannot continue until you fix this that and that`

and from your description it is 100% a warning. Okay that we will ignore other formulas, but we still let the feature go on and complete. It's not a problem that we don't have 

I'd rather use a warning message then, clarifying also the expected behavior: `Multiple Group formulas detected. Only one formula for "Bind" can be used on each system! The first "Bind" formula will be used, others will be ignored.`.

Reply:
Okay, that makes sense. Thanks!

Reply:
Are we sure we know what "first" formula is? From what I remember we don't sort group list and do not operate on ordered list. So is it first group assigned? Or is it first group alphabetically?
When I look in retail terminal group membership I see:

```
# cat pillar_sbranch1.Intel-Genuine-c557_group_memberships.yml
addon_group_types:
- salt_entitled
group_ids:
- 11
- 9
- 12```

Where this machine was first assigned to group 11, then 12 and last 9. If I had the same formula for each group. Which one would be "the first"?

Reply:
> In current situation I consider it as an error. Merging data from multiple groups is silently disabled at the moment (see [uyuni/suma_minion.py#L155](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/modules/pillar/suma_minion.py?rgh-link-date=2020-07-15T17%3A25%3A27Z#L155) ), and furthermore there is not a stable way to tell what group formula will "win" and be loaded.
> 
> However there is an exception for cluster-formula, @mbologna ?

Yes, that is correct: `caasp-management-settings-formula` has the type of `cluster-formula`

Reply:
You are right  @aaannz, I would not put "first" either. Which group is parsed first is not really intuitive. In your case it would be group 11. Below is my observation:

I just created three groups (1-3) on a fresh SUMA  4.1 Server. I added the groups in to a Client in the order `3`, `2`, `1` and then added `bind` formulas to `1`, then to `2` and lastly to `3`. In the end, the minion shows three "Bind" formulas, all with the data from `2`. This is aligned with the order in `pillar_minion-1.tf.local_group_memberships.yml`, but not what a user would expect. 

I did not look into where the order is decided, why I the `2` group is the first in the yaml and if that process is deterministic. The "good thing" is that all formulas on `/rhn/manager/systems/details/formula/` show the same content, so at least the user can see what is going to be applied.

So I think it is good enough to tell the user that multiple formulas are not taken into account. I don't think we have the correct group id easily available in the `FormulaForm.js` to show it.